experimental/web/sample_dynamic/CMakeLists.txt

```diff
@@ -33,11 +33,14 @@ target_link_libraries(${_NAME}
 
 target_link_options(${_NAME} PRIVATE
   # https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html#interacting-with-code-ccall-cwrap
-  "-sEXPORTED_FUNCTIONS=['_load_program']"
+  "-sEXPORTED_FUNCTIONS=['_setup_sample', '_cleanup_sample', '_load_program', '_unload_program', '_call_function']"
   "-sEXPORTED_RUNTIME_METHODS=['ccall','cwrap']"
   #
   "-sASSERTIONS=1"
   #
+  # Programs loaded dynamically can require additional memory, so allow growth.
+  "-sALLOW_MEMORY_GROWTH"
```

Comment:
this doesn't work with threading right? I think it's fine to have builds with and without threading vs making an uber build to start, so should be something we can work around. probably going to have to do some deeper design work to make things production ready though - we don't want to alloc 100MB to copy in 100MB of model weights to then copy back out to WebGPU etc.

Reply:
Yes, this dynamic sample does not work with threading, because Emscripten's load time dynamic linking is experimental and requires `MAIN_MODULE` / `SIDE_MODULE` (all through emcc, while our executables are produced by LLVM and wasm-ld directly), [see Emscripten docs](https://emscripten.org/docs/compiling/Dynamic-Linking.html#pthreads-support). This is one of the main reasons to move away from Emscripten, IMO: https://github.com/google/iree/issues/8327.

This particular `ALLOW_MEMORY_GROWTH` flag works with threading, but [the Emscripten docs](https://emscripten.org/docs/porting/pthreads.html#special-considerations) say it comes with a performance cost, particularly if the JS does large amounts of memory reads and writes. I still have your comment from https://github.com/google/iree/pull/8291#discussion_r804271876 to investigate, to see what is actually taking up this memory (runtime overhead / model weights / emscripten silliness).